v-next/hardhat/src/internal/builtin-plugins/network-manager/request-handlers/hanlders-array.ts

```diff
@@ -0,0 +1,124 @@
+import type { RequestHandler } from "./types.js";
+import type {
+  HDAccountsUserConfig,
+  HttpNetworkAccountsUserConfig,
+} from "../../../../types/config.js";
+import type {
+  ChainType,
+  NetworkConnection,
+} from "../../../../types/network.js";
+
+import { numberToHexString } from "@ignored/hardhat-vnext-utils/hex";
+
+import { AutomaticSenderHandler } from "./handlers/accounts/automatic-sender-handler.js";
+import { FixedSenderHandler } from "./handlers/accounts/fixed-sender-handler.js";
+import { HDWalletHandler } from "./handlers/accounts/hd-wallet-handler.js";
+import { LocalAccountsHandler } from "./handlers/accounts/local-accounts.js";
+import { ChainIdValidatorHandler } from "./handlers/chain-id/chain-id-handler.js";
+import { AutomaticGasHandler } from "./handlers/gas/automatic-gas-handler.js";
+import { AutomaticGasPriceHandler } from "./handlers/gas/automatic-gas-price-handler.js";
+import { FixedGasHandler } from "./handlers/gas/fixed-gas-handler.js";
+import { FixedGasPriceHandler } from "./handlers/gas/fixed-gas-price-handler.js";
+
+/**
+ * This function returns an handlers array based on the values in the NetworkConnection and NetworkConfig.
+ * The order of the handlers, if all are present, is: chain handler, gas handlers (gasPrice first, then gas), sender handler and accounts handler.
+ * The order is important to get a correct result when the handlers are executed.
+ */
+export async function createHandlersArray<
+  ChainTypeT extends ChainType | string,
+>(networkConnection: NetworkConnection<ChainTypeT>): Promise<RequestHandler[]> {
+  const requestHandlers = [];
+
+  const networkConfig = networkConnection.networkConfig;
+
+  if (networkConfig.type === "http" && networkConfig.chainId !== undefined) {
+    requestHandlers.push(
+      new ChainIdValidatorHandler(
+        networkConnection.provider,
+        networkConfig.chainId,
+      ),
+    );
+  }
+
+  if (
+    networkConfig.gasPrice === undefined ||
+    networkConfig.gasPrice === "auto"
+  ) {
+    // If you use a hook handler that signs locally, you are required to
+    // have all the transaction fields available, including the
+    // gasPrice / maxFeePerGas & maxPriorityFeePerGas.
+    //
+    // We never use those when using EDR Network, as we sign within the
+    // EDR Network itself. This means that we don't need to provide all the
+    // fields, as the missing ones will be resolved there.
+    //
+    // EDR Network handles this in a more performant way, so we don't use
+    // the AutomaticGasPrice for it.
+    if (networkConfig.type === "http") {
```

Comment:
Is this the if condition that can be removed because also supported in EDR networks?
See related issue here: https://github.com/NomicFoundation/hardhat/issues/5827

Reply:
Let's remove this one, but not the one in line 92. Let's let EDR handle the signing of transactions from its own accounts.

Reply:
Done [here](https://github.com/NomicFoundation/hardhat/pull/5994/commits/2fa55f6cc21d79ec78f11dca321ff12057613355)